fix: ensure sane accounts in test data

jniles opened the pull request:
This commit ensures that we have sane accounts in our test dataset by changing a title account to an income account.

![image](https://user-images.githubusercontent.com/896472/91724015-8eb2ef80-eb94-11ea-8f97-4ec256b685a8.png)

Also detects title accounts in the trial balance.

Closes #4846.

jniles committed:
fix: ensure sane accounts in test data

This commit ensures that we have sane accounts in our test dataset by
changing a title account to an income account.

Closes #4946.

lomamech commented on test/data.sql:
```diff
@@ -2930,10 +2930,10 @@ INSERT INTO `voucher_item` (`uuid`, `account_id`, `debit`, `credit`, `voucher_uu
-  (HUID('953defee-b1b2-11e8-ac16-3b802412d42d'), 200, 0, 200, @second_voucher, NULL, NULL),
+  (HUID('953defee-b1b2-11e8-ac16-3b802412d42d'), 242, 0, 200, @second_voucher, NULL, NULL),
   (HUID('97ad9a22-b1b2-11e8-acba-6b85fd27d57d'), 125, 300, 0, @third_voucher, @cash_payment, HUID('2f0b966c-b1b3-11e8-8dd2-1715a827ad9b')),
   (HUID('8ba0571a-b1b2-11e8-8688-fb9547361273'), 117, 0, 300, @third_voucher, NULL, NULL),
```
On this line, the account identifier 117 corresponds to that of the account [165 - Deposits and  Guarantees received]

lomamech reviewed (commented):
Here is some transaction on the test database that uses title accounts
- TPC1 with Id 117 for [165 - Depots et Cautionnement recus]
- TPC1 with Id 125 for  [1861 - Comptes de liaison charges]
- TPA42 with 200 Id  [601 - ACHATS DE MARCHANDISES ]

I think it will also be necessary to revisit the accounts used in different Integrations and E2E tests

jniles committed:
fix(trial balance): catch title accounts w/ balances

Adds a check to ensure that transactions are not posted with title
accounts that have balances.

Closes #4846.
